### PR TITLE
[ARTS-382] Specific environment deployments

### PR DESCRIPTION
**Pull request by mdp0**
## Description
Add a workflow dispatch trigger to existing workflow, accepting a path to a trial definition file as an argument. Deploy only the requested trial if the argument is present

### Changes
- [ARTS-382] - Add workflow_dispatch with argument

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-382]: https://ndph-arts.atlassian.net/browse/ARTS-382

### CODE REVIEW

**Review by darrengsmithson — COMMENTED** [review]
All good but general comment..
Does the defPath need to be absolute i.e mts-trial-deployment-config/trials/qa1/definition.json? I did a similar thing which wasnt merged into main where the inout was a 'name' - i.e QA - https://github.com/NDPH-ARTS/mts-trial-deployment-config/blob/feature/arts-307-manual-workflow/.github/workflows/terraform-plan-apply.yml

File: trials/qa1/definition.json [hunk 1]
```diff
@@ -7,32 +7,32 @@
     "spring_config_label": "main",
     "site_service": {
         "image_name": "ghcr.io/ndph-arts/site-service",
-        "image_tag": "0e310062adb3569db688214ef4ad7273a7918b4d"
```
Comment (eladiw) [inline trials/qa1/definition.json]: qa1, should not be changed. it should have a specific tag

File: .github/workflows/terraform-plan-apply.yml [hunk 2]
```diff
@@ -2,7 +2,14 @@ name: Terraform Plan and Apply for Trials
 
 on:
   push:
-
+    paths-ignore:
```
Comment (eladiw) [inline .github/workflows/terraform-plan-apply.yml]: L5 and L6 were removed in a previous pr

**Review by darrengsmithson — DISMISSED** [review]

**Review by eladiw — CHANGES_REQUESTED** [review]
qa1 can't be changed. by changing it you affect the qa1 deployment env. so it must be reverted.

File: .github/workflows/terraform-plan-apply.yml (hunk 2 above)
Comment (mdp0) [reply .github/workflows/terraform-plan-apply.yml]: Removed

File: trials/qa1/definition.json (hunk 1 above)
Comment (mdp0) [reply trials/qa1/definition.json]: reverted

File: .github/workflows/terraform-plan-apply.yml [hunk 3]
```diff
@@ -27,25 +32,29 @@ jobs:
           set -o nounset
           # set -o xtrace
 
-          git fetch origin main:localmain
+          if [[ -z "${{github.event.inputs.trialDefinitionPath}}}" ]]; then
```
Comment (eladiw) [inline .github/workflows/terraform-plan-apply.yml:38]: can you add a comment such as:
"if a specific trial was requested, we deploy it. meaning this was an external trigger (not a pr). other wise we deploy only changed files"

**Review by eladiw — CHANGES_REQUESTED** [review]
add some documentation.
also, do you want to include a nightly trial in this pr?

File: trials/main-nightly/definition.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    image_name: ghcr.io/ndph-arts/gateway-server
+    image_tag: main-nightly
+  discovery_service:
+    image_name: ghcr.io/ndph-arts/discovery-server
+    image_tag: main-nightly
+  config_server:
+    image_name: ghcr.io/ndph-arts/config-server
+    image_tag: main-nightly
+    git_uri: https://github.com/NDPH-ARTS/mts-trial-deployment-config.git
+    search_paths: trials/qa1/services, trials/qa1/services/*
```
Comment (eladiw) [inline trials/main-nightly/definition.yml]: main-nightly will be the trial name so only alpha numeric value.
so need to rename the trial name and dir: nightly, or mainnightly

also the search paths need to point to the new directory and not qa1

**Review by eladiw — CHANGES_REQUESTED** [review]
some comments

File: trials/main-nightly/definition.yml (hunk 4 above)
Comment (mdp0) [reply trials/main-nightly/definition.yml]: Renamed

File: .github/workflows/terraform-plan-apply.yml (hunk 3 above)
Comment (mdp0) [reply .github/workflows/terraform-plan-apply.yml:38]: Added comment

**Review by eladiw — DISMISSED** [review]
in the second half of the task. please delete all artsxxx.yml files

**Review by darrengsmithson — APPROVED** [review]

**Review by eladiw — APPROVED** [review]

**Pull request merged**